src/browser/modules/Sidebar/project-files.utils.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ * This file is part of Neo4j.
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import remote from 'services/remote'
+import { SLASH, CYPHER_FILE_EXTENSION } from 'shared/services/export-favorites'
+import { pick } from 'lodash-es'
+import uuid from 'uuid'
+import { split, trim, head, startsWith } from 'lodash-es'
+
+import {
+  IProjectFile,
+  IFavorite,
+  IProjectFilesResult,
+  GET_PROJECT_FILES,
+  IAddProjectFile,
+  IRemoveProjectFile
+} from './project-files.constants'
+import {
+  ApolloCache,
+  FetchResult,
+  NormalizedCacheObject,
+  Reference
+} from '@apollo/client'
+
+export const ProjectFilesQueryVars = {
+  projectId: 'project-03688c10-a811-4c0c-85d4-581e88c2183a', // @todo: hardcoded for now
+  filterValue: CYPHER_FILE_EXTENSION
+}
+
+export const ProjectFileMutationVars = (
+  filePath: string
+): { projectId: string; filePath: string } => ({
+  ...pick(ProjectFilesQueryVars, ['projectId']),
+  filePath
+})
+
+export const mapProjectFileToFavorites = async ({
+  downloadToken,
+  name,
+  directory
+}: IProjectFile): Promise<IFavorite> => ({
+  id: uuid.v4(),
+  name,
+  directory,
+  path: directory.startsWith('.') ? SLASH : `${SLASH}${directory}`, // @todo: have to add SLASH to show files/folders
```

Comment:
Do these kinds of slash related things work when relate runs on windows?

Reply:
its only for use in a saved-scripts util to display correctly, can be looked into at a later date. I am passing down name and directory and thats the main properties needed.

Reply:
Ah okay! It'd be nice if we could test this branch on windows before merging anyhow, I feel like there might be dragons

Reply:
for sure

Reply:
This path is provided by the backend I believe to we could see a difference on Desktop windows installations

Reply:
Ill catch up with you @huboneo